### PR TITLE
fix: making track optional

### PR DESCRIPTION
**Pull request by maze-runnar**
related https://github.com/fossasia/open-event-frontend/pull/5470

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/sessions.py [hunk 1]
```diff
@@ -59,8 +59,9 @@ def before_post(self, args, kwargs, data):
         ):
             raise ForbiddenError({'pointer': ''}, "Sessions are disabled for this Event")
 
+        excluded = ['track']
```
Comment (iamareebjamal) [inline app/api/sessions.py:62]: But we want it to be required when the organizers have marked it required

Comment (maze-runnar) [reply app/api/sessions.py:62]: how to check if any custom form field is required?

Comment (maze-runnar) [reply app/api/sessions.py:62]: I have a doubt, when other custom form fields are made required they only have frontend validation or for each required field the validation in server

Comment (maze-runnar) [reply app/api/sessions.py:62]: Because for track also, frontend validation is working while it is made required.

Comment (iamareebjamal) [reply app/api/sessions.py:62]: They have server validation as well

Comment (maze-runnar) [reply app/api/sessions.py:62]: how to check that track is required or not on server side?

Comment (iamareebjamal) [reply app/api/sessions.py:63]: By not passing track in excluded

File: app/api/sessions.py [hunk 2]
```diff
@@ -59,8 +60,16 @@ def before_post(self, args, kwargs, data):
         ):
             raise ForbiddenError({'pointer': ''}, "Sessions are disabled for this Event")
 
+        
+        custom_fields = db.session.query(CustomForms).filter_by(event_id = data['event']).all()
+        for i in custom_fields:
+            if(i.field_identifier == "track" and i.is_required == True):
+                excluded = []
+            elif(i.field_identifier == "track" and i.is_required == False):
+                excluded = ['track']
```
Comment (iamareebjamal) [inline app/api/sessions.py:69]: Shouldn't be needed now

Comment (maze-runnar) [reply app/api/sessions.py]: only removing track from required relation is not working.

File: app/api/sessions.py [hunk 3]
```diff
@@ -59,8 +60,15 @@ def before_post(self, args, kwargs, data):
         ):
             raise ForbiddenError({'pointer': ''}, "Sessions are disabled for this Event")
 
+        
+        custom_fields = db.session.query(CustomForms).filter_by(event_id = data['event']).all()
+        excluded = ['track']
+        for i in custom_fields:
+            if(i.field_identifier == "track" and i.is_required == True):
+                excluded = []
```
Comment (iamareebjamal) [inline app/api/sessions.py:68]: excluded shouldn't be needed now

Comment (maze-runnar) [reply app/api/sessions.py:68]: means only exclude = ["track"] is enough?

File: app/api/sessions.py (hunk 2 above)
Comment (iamareebjamal) [reply app/api/sessions.py]: What's the error?

Comment (maze-runnar) [reply app/api/sessions.py]: sorry, I had made a mistake. It's working corretly.

File: app/api/sessions.py [hunk 4]
```diff
@@ -60,8 +60,7 @@ def before_post(self, args, kwargs, data):
             raise ForbiddenError({'pointer': ''}, "Sessions are disabled for this Event")
 
         data['complex_field_values'] = validate_custom_form_constraints_request(
-            'session', self.schema, Session(event_id=data['event']), data
-        )
+            'session', self.schema, Session(event_id=data['event']), data)
```
Comment (iamareebjamal) [inline app/api/sessions.py]:
```suggestion
            'session', self.schema, Session(event_id=data['event']), data
        )
```

**Pull request merged**